exporters/committime/collector_image.py

```diff
@@ -0,0 +1,214 @@
+        repo_url=("io.openshift.build.source-location", False),
+        committer=("io.openshift.build.commit.author", False),
+    )
+
+    def __init__(self, kube_client, apps, date_format):
+        super().__init__(
+            kube_client,
+            None,
+            None,
+            None,
+            apps,
+            "Image",
+            date_format,
+        )
+
+    def commit_metric_from_image(self, app: str, image, errors: list) -> CommitMetric:
+        """
+        Create a CommitMetric from image.openshift.io/v1 information or Image annotation
+
+        Some of the information such as image sha is gathered from the Image metadata,
+        which we expect to always exists.
+
+        For the commit time Image type exporter only commit time is required, however
+        additional data is also collected such as commit hash.
+
+        commit time which is converted to the commit timestamp is gathered from the
+        Image Label, which normally is populated from the Docker build process as
+        described in https://docs.openshift.com/online/pro/dev_guide/builds/build_output.html#output-image-labels
+
+        If such information is missing from the Image Label there is a way to collect
+        this data using annotations in similar way build annotations works.
+
+        """
+        metric = CommitMetric(app)
+        image_labels = None
+
+        # If exists get all Labels that were produced from Docker build process
+        with collect_bad_attribute_path_error(errors, False):
+            image_labels = get_nested(
+                image, "dockerImageMetadata.Config.Labels", name="image"
+            )
+
+        # Get general data from image
+        for attr_name, (path, required) in ImageCommitCollector._IMAGE_MAPPING.items():
+            with collect_bad_attribute_path_error(errors, required):
+                value = get_nested(image, path, name="image")
+                setattr(metric, attr_name, value)
+
+        # First get metrics within Labels
+        if image_labels:
+            for attr_name, (
+                path,
+                required,
+            ) in ImageCommitCollector._DOCKER_LABEL_MAPPING.items():
+                with collect_bad_attribute_path_error(errors, required):
+                    value = image_labels.get(path)
+                    if value:
+                        if (
+                            path
+                            == ImageCommitCollector._DOCKER_LABEL_MAPPING.get(
+                                "commit_time"
+                            )[0]
+                        ):
+                            logging.debug(
+                                "Commit time for image %s provided by '%s' label: %s",
+                                metric.image_hash,
+                                path,
+                                value,
+                            )
+
+                        setattr(metric, attr_name, value)
```

Comment:
I think we could simplify this section, and avoid some issues in here too.

There's no `get_nested` calls here, so the `collect_bad_attribute_path_error` won't collect anything. Also, we get up to 5 levels of indentation deep from the function body. That makes reading it confusing, make it harder to pass the `mccabe` check.

Might I propose:

```python
attribute_mapping = ImageCommitCollector._DOCKER_LABEL_MAPPING.items() if image_labels else []
for attr_name, (path, required) in attribute_mapping:
    with collect_bad_attribute_path_error(errors, required):
        value = get_nested(image_labels, path)
        setattr(metric, attr_name, value)
        if attr_name == "commit_time":
            logging.debug(
                "Commit time for image %s provided by '%s' label: %s",
                metric.image_hash,
                path,
                value,
            )
```